community/kernel/src/main/java/org/neo4j/GraphDatabaseBuilder.java

```diff
@@ -0,0 +1,395 @@
+        return self();
+    }
+
+    /**
+     * Set the default cypher planner.
+     *
+     * @param name The name of the cypher planner to use by default
+     * @return this builder
+     */
+    public BUILDER usingCypherPlanner( String name )
+    {
+        params.put( GraphDatabaseSettings.cypher_planner.name(), name );
+        return self();
+    }
+
+    /**
+     * Set the default cypher runtime.
+     *
+     * @param name The name of the cypher runtime to use by default
+     * @return this builder
+     */
+    public BUILDER usingCypherRuntime( String name )
+    {
+        params.put( GraphDatabaseSettings.cypher_runtime.name(), name );
+        return self();
+    }
+
+    /**
+     * Set the size of the cypher query cache (default is 1000).
+     *
+     * @param size the size of the cypher query cache
+     * @return this builder
+     */
+    public BUILDER withCypherQueryCacheSize( int size )
+    {
+        params.put( GraphDatabaseSettings.query_cache_size.name(), String.valueOf( size ) );
+        return self();
+    }
+
+    /**
+     * Set the minimum age of a query plan before the query is considered for replanning.
+     *
+     * @param age the age (in seconds) of a query plan before the query is considered for replanning
+     * @return this builder
+     */
+    public BUILDER withCypherReplanInterval( int age )
+    {
+        params.put( GraphDatabaseSettings.cypher_min_replan_interval.name(), String.valueOf( age ) );
+        return self();
+    }
+
+    /**
+     * Disable the use of file URIs (i.e. <code>file://</code>) in Cypher <code>LOAD CSV</code> clauses.
+     *
+     * @return this builder
+     */
+    public BUILDER disallowCypherLoadCSVFileURIs()
+    {
+        params.put( GraphDatabaseSettings.allow_file_urls.name(), String.valueOf( false ) );
+        return self();
+    }
+
+    /**
+     * Set the threshold size before the internal log (messages.log) is rotated.
+     *
+     * @param bytes the threshold in bytes
+     * @return this builder
+     */
+    public BUILDER withInternalLogRotationThreshold( long bytes )
+    {
+        params.put( GraphDatabaseSettings.store_internal_log_rotation_threshold.name(), String.valueOf( bytes ) );
+        return self();
+    }
+
+    /**
+     * Set the delay since the last rotation before the internal log (messages.log) can be rotated again.
+     *
+     * @param delay the delay in seconds
+     * @return this builder
+     */
+    public BUILDER withInternalLogRotationDelay( int delay )
+    {
+        params.put( GraphDatabaseSettings.store_internal_log_rotation_delay.name(), String.valueOf( delay ) );
+        return self();
+    }
+
+    /**
+     * Set the maximum number of archived internal logs to retain.
+     *
+     * @param count the maximum number of archived internal logs to retain
+     * @return this builder
+     */
+    public BUILDER withInternalLogMaxArchives( int count )
```

Comment:
There's been a couple of suggestions that only the most common options should be methods. I'm not sure I buy that though, because then the rest have to be set via string interpretation -- which doesn't really make sense in a programatic API, unless you're trying to hide those as internal-only options. It does make sense in server, where values are read from a properties file. To save on argument though, let's merge this with a minimal set while the discussion continues (I can move the additional ones to a new PR). Is it just this method you're questioning?